### PR TITLE
🌡️ / 🥫

### PR DESCRIPTION
**Pull request by ActuallyTaylor**
Added the ability for the TMU too send its temperature data over CAN

### CODE REVIEW

**Comment by ActuallyTaylor** [comment]
[TMU CANopen Setup](https://trello.com/c/22Fp63k4/126-tmu-canopen-setup)

File: include/TMU.hpp [hunk 1]
```diff
@@ -1,13 +1,167 @@
...
+         * 0. The number of sub indxes.
+         * 1. The COBID for the transmitting node
+         * 2. The transmission trigger 0xFE is asynchronous
+         * 3. The inhibit time
+         * 4. The event timer
+         */
+        {
+            .Key = CO_KEY(0x1800, 0, CO_UNSIGNED8 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 5},
```
Comment (mjmagee991) [inline include/TMU.hpp]:
```suggestion
            .Data = (uintptr_t) 5,},
```
If you have a comma here in each element in the list, the autoformatter is much happier

File: src/TMU.cpp [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
 
-namespace BOARD_NAME {}
+namespace TMU {
+CO_OBJ_T* TMU::getObjectDictionary() {
+    return &objectDictionary[0];
+}
+
+uint16_t TMU::getObjectDictionarySize() const {
+    return OBJECT_DICTIONARY_SIZE;
+}
```
Comment (mjmagee991) [inline src/TMU.cpp:10]:
```suggestion
}

```

File: targets/DEV1-TMU/CMakeLists.txt [hunk 3]
```diff
@@ -1,7 +1,8 @@
 include(${EVT_CORE_DIR}/cmake/evt-core_build.cmake)
 
-project(BIKE_NAME-BOARD_NAME)
+project(TMUmain)
```
Comment (mjmagee991) [inline targets/DEV1-TMU/CMakeLists.txt]:
```suggestion
project(DEV1-TMU)
```

File: targets/DEV1-TMU/CMakeLists.txt [hunk 4]
```diff
@@ -1,7 +1,8 @@
...
 cmake_minimum_required(VERSION 3.15)
 
 make_exe(${PROJECT_NAME} main.cpp)
 target_link_libraries(${PROJECT_NAME} PUBLIC ${BOARD_LIB_NAME})
+
```
Comment (mjmagee991) [inline targets/DEV1-TMU/CMakeLists.txt]:
```suggestion
```

File: targets/DEV1-TMU/main.cpp [hunk 5]
```diff
@@ -4,26 +4,172 @@
...
 int main() {
     // Initialize system
     IO::init();
 
     // Setup UART
     IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+    log::LOGGER.setUART(&uart);
+
+    // Initialize the timer
+    DEV::Timerf302x8 timer(TIM2, 100);
```
Comment (mjmagee991) [inline targets/DEV1-TMU/main.cpp]: Update the EVT-core submodule and make this platform agnostic

File: targets/DEV1-TMU/main.cpp [hunk 6]
```diff
@@ -4,26 +4,172 @@
...
+    timer.stopTimer();
+
+    TMU::TMU tmu{};
```
Comment (mjmagee991) [inline targets/DEV1-TMU/main.cpp]:
```suggestion
    TMU::TMU tmu;
```

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
A couple formatting issues and one platform problem

**Review by ishan-renjen — APPROVED** [review]

File: src/TMU.cpp [hunk 7]
```diff
@@ -1,3 +1,12 @@
 #include <TMU.hpp>
 
-namespace BOARD_NAME {}
+namespace TMU {
+CO_OBJ_T* TMU::getObjectDictionary() {
+    return &objectDictionary[0];
```
Comment (mjmagee991) [inline src/TMU.cpp]:
```suggestion
    return objectDictionary;
```

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
One minor detail

**Review by mjmagee991 — APPROVED** [review]
Nice work

**Pull request merged**